src/Behat/Transliterator/SyncTool.php

```diff
@@ -0,0 +1,215 @@
+    {
+        if ($this->itemIndex >= 16) {
+            $this->phpTable = trim($this->phpTable);
+            $this->phpTable .= PHP_EOL;
+            $this->itemIndex = 0;
+        }
+        ++$this->itemIndex;
+
+        $item = new StringValue($item);
+        if ($item->starts('\x') || $item->starts('\n')) {
+            $this->phpTable .= '"' . $item . '", ';
+            $this->nonQuestionBoxFound = true;
+        } else {
+            // TODO check if this hack should be removed for chinese letters
+            if ($item->value === '[?] ') {
+                $item->value = '[?]';
+            }
+            //
+
+            if ($item->value !== '[?]') {
+                $this->nonQuestionBoxFound = true;
+            }
+
+            $this->phpTable .= "'" . str_replace(array('\\', '\''), array('\\\\', '\\\''), $item) . "', ";
+        }
+    }
+
+    private function tokenizePerlTable($content)
+    {
+        $tokens = $this->tokenizer->tokenize($content);
+
+        $expression = $this->renderer->getExpression($tokens);
+        $statement = $expression->getStatement();
+        /** @var Parsed[] $binds */
+        $binds = $expression->getBinds();
+
+        $parser = new StringParser($statement);
+        $block = (string)$parser->inner('$Text::Unidecode::Char[', ']');
+        if (!$block) {
+            throw new \Exception('Block not found');
+        }
+        $this->block = $this->renderer->getExpression($binds[$block])->getStatement();
+
+        $itemsBind = (string)$parser->inner('[', ']');
+
+        if (!$itemsBind) {
+            $items = array();
+        }
+        else {
+            $items = $binds[$itemsBind];
+        }
+
+        return $items;
+    }
+
+    private function parsePerlTable($content)
+    {
+        $items = $this->tokenizePerlTable($content);
+
+        $phpFilePath = __DIR__ . '/data/' . substr($this->block, 1) . '.php';
+        if (!$items) {
+            $this->removePhpCharTable($phpFilePath, 'Empty char table for block ' . $this->block);
+            return;
+        }
+
+        $this->phpTable = <<<PHP
+<?php
+\$UTF8_TO_ASCII[$this->block] = array(
+
+PHP;
+
+        $itemsExpression = $this->renderer->getExpression($items);
+        $itemsStatement = $itemsExpression->getStatement();
+        $itemsBinds = $itemsExpression->getBinds();
+
+        $itemsStatement = explode(',', $itemsStatement);
+        $this->itemIndex = 0;
+        $this->nonQuestionBoxFound = false;
+        foreach ($itemsStatement as $item) {
+            $item = trim($item);
+            if (!$item) {
+                break;
+            }
+
+            if (isset($itemsBinds[$item])) {
+                /** @var Token $token */
+                $token = $itemsBinds[$item];
+                $item = $token->unEscapedContent;
+            }
+
+            $this->pushItem($item);
+        }
+
+        if ($this->nonQuestionBoxFound) {
+            $this->phpTable = trim($this->phpTable) . PHP_EOL . ');' . PHP_EOL;
```

Comment:
please use ``\n`` in the content of the file being written instead of ``PHP_EOL``, so that we don't end up with mixed line endings when running the script on windows